lib/rpmsg/rpmsg_virtio.c

```diff
@@ -414,6 +432,40 @@ static int rpmsg_virtio_send_offchannel_nocopy(struct rpmsg_device *rdev,
 	return len;
 }
 
+static int rpmsg_virtio_release_tx_buffer(struct rpmsg_device *rdev, void *rxbuf)
+{
+	struct rpmsg_virtio_device *rvdev;
+	struct rpmsg_hdr *rp_hdr;
+	int status = RPMSG_SUCCESS;
+	uint16_t idx;
+
+	rvdev = metal_container_of(rdev, struct rpmsg_virtio_device, rdev);
+
+	metal_mutex_acquire(&rdev->lock);
+
+	if (rvdev->desc_recycler_cnt == RPMSG_RECYCLER_SIZE) {
+		status = RPMSG_ERR_NO_MEM;
+		goto ret_status;
+	}
+
+	rp_hdr = RPMSG_LOCATE_HDR(rxbuf);
```

Comment:
since rxbuf is free, why not reuse rxbuf to form a free link list, so:

1. We can remove desc_recycler and the limitation of RPMSG_RECYCLER_SIZE
2. Avoid to increase the size of struct rpmsg_virtio_device

Reply:
yes would be nice to be able to remove limitation. But here rxbuf is pointing to the rpmsg payload, so I don't see how I can reuse it as I should give the pointer on `rpmsg_get_tx_payload_buffer`.
Please, could you detail what you have in mind, if I missed something?

Reply:
we can directly cast rxbuf to desc_recycler_t * and link into the list since nobody use rxbuf after this point.

Reply:
Got it!
Your proposal is to reuse the rpmsg payload to temporay store the desc_recycler_t structure.
That's seems a good idea, I will rework the PR to implement in this way.
Thanks for the advice!